tests/integration-tests/tests/multi-user/test_multi_user.py

```diff
@@ -0,0 +1,321 @@
+        custom_ami=custom_ami,
+        bucket_name=bucket_name,
+        active_directory_policy_arn=active_directory_policy_arn,
+        lambda_function_name=function_name,
+    )
+    cluster = clusters_factory(cluster_config, raise_on_error=False)
+
+    _assert_head_node_is_running(region, cluster)
+
+    active_directory_admin_user = "administrator"
+    assert_aws_identity_access_is_correct(
+        cluster=cluster, users_allow_list={"root": True, "pcluster-admin": True, active_directory_admin_user: False}
+    )
+
+    # TODO add assertions about ssh access and job submission
+    # assert_ssh_login(cluster=cluster, username=active_directory_admin_user, ssh_key=private_key_path)
+
+
+@pytest.mark.dimensions("us-east-1", "c5.xlarge", "*", "*")
+@pytest.mark.usefixtures("instance", "scheduler")
+def test_create_multi_user_with_opsworks(
+    region,
+    os,
+    pcluster_config_reader,
+    s3_bucket_factory,
+    clusters_factory,
+    architecture,
+    vpc_stack,
+    test_datadir,
+    policy_factory,
+    role_factory,
+    instance_profile_factory,
+    user_factory,
+    opsworks_stack_factory,
+    opsworks_user_profile_factory,
+):
+    """Test the creation of a cluster whose users are managed with OpsWorks."""
+
+    # Create IAM resources required by OpsWorks stack
+    opsworks_service_role_policy_arn = create_policy_from_file(
+        policy_factory, f"{test_datadir}/opsworks_service_role_policy.json"
+    )
+    _, opsworks_service_role_arn = role_factory(trusted_service="opsworks", policies=[opsworks_service_role_policy_arn])
+    opsworks_instance_profile_arn = instance_profile_factory()
+
+    # Create OpsWorks stack
+    vpc_id = vpc_stack.cfn_outputs["VpcId"]
+    subnet_id = vpc_stack.cfn_outputs["PublicSubnetId"]
+    opsworks_stack_id = opsworks_stack_factory(
+        vpc_id, subnet_id, opsworks_service_role_arn, opsworks_instance_profile_arn
+    )
+
+    # Create resource bucket and upload scripts
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    bucket.upload_file(str(test_datadir / "post_install.sh"), "post_install.sh")
+
+    # Create cluster
+    custom_ami = retrieve_latest_ami(region, os, ami_type="pcluster", architecture=architecture)
+    cluster_config = pcluster_config_reader(
+        custom_ami=custom_ami, bucket_name=bucket_name, opsworks_stack_id=opsworks_stack_id
+    )
+    cluster = clusters_factory(cluster_config, raise_on_error=False)
+
+    _assert_head_node_is_running(region, cluster)
+
+    # The opsworks-agent process is run as aws user, so it requires the access to IMDS
+    configure_imds_access(cluster, {"aws": True})
+
+    # Create new user credentials
+    key = RSA.generate(2048)
+    private_key = key.export_key()
+    private_key_path = f"{test_datadir}/private.key"
+    with (open(private_key_path, "wb")) as file:
+        file.write(private_key)
+    public_key = key.publickey().export_key().decode("utf-8")
```

Comment:
it would be nice to wrap this in a function so that it can be used by all the multi-user tests that need to generate a keypair. In the OpenLDAP test case I used subprocess to call ssh-keyegn, but avoiding the external dependency seems cleaner for sure.